packages/squiggle-lang/src/public/SqProject/index.ts

```diff
@@ -246,26 +251,34 @@ export class SqProject {
       })
     );
 
-    const [bindings, exports] = (["bindings", "exports"] as const).map(
-      (field) =>
-        new SqDict(
-          internalOutputR.value[field],
-          new SqValueContext({
-            project: this,
-            sourceId,
-            source,
-            ast: astR.value,
-            valueAst: astR.value,
-            valueAstIsPrecise: true,
-            path: new SqValuePath({
-              root: "bindings",
-              items: [],
-            }),
-          })
-        )
-    );
+    const [bindings, exports, imports] = (
+      ["bindings", "exports", "imports"] as const
+    ).map((field) => {
+      const innerDict =
+        field === "imports"
+          ? internalOutputR.value.externals.explicitImports
+          : internalOutputR.value[field];
+      const dict = new SqDict(
+        field === "exports"
```

Comment:
These ternaries are complicated, I'd suggest this:
```typescript
const bindings = makeTopDict(internalOutputR.value.bindings);
const exports = makeTopDict(internalOutputR.value.exports.mergeTags({ name: sourceId }));
const imports = makeTopDict(internalOutputR.value.externals.explicitImports);
```

With `makeTopDict` containing the rest of the common code.